src/app/modules/item/components/item-chapter-view/group-situation-chapter-view/user-progress/user-progress.component.ts

```diff
@@ -1,6 +1,8 @@
-import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { Component, Input, OnChanges, OnDestroy, SimpleChanges } from '@angular/core';
```

Comment:
I am sorry but I miss some important design "flaws" in the previous PRs.

1) This component is stored inside the first component. This is probably something done by the design, but that makes the project become a mess. There is no hierarchy between component. Oh here there is even an addition layer of hierarchy.
And the way it was organized was supposed to be: the actual content with data logic etc should be in "/pages" while independent component just for display should be "/components". So here, the part with the matrix should be in pages while the cell component should be in "component".

2) This should be the page (so the component of the matrix somehow) handling what "info" popup and permission dialog is open etc. There is only one open at a time anyway... It is like implementing is in the button component.  The user progress should be a design only thing.

3) (related with 2) I missed it but possibly that explain why this matrix is so slow, you have a matrix on possibly 15 columns, 1500 rows, so 22500 components, each storing their item, group, permissions, a state, possibly a rx observable, ... 

4) Each of these possibly 22500 components is filled with "find", which iterate over every element of a very long list to find its value. You could create the datastructure first and fill it while reading the data once. Using a O(n^2) algorithm instead of O(n) may have a big impact when that's related with thousands of cells.

Maybe all this may be fixed different PR when this PR is merged(?)